### ISSUE TITLE
Implement missile system for better tower/mob spells

### ISSUE THREAD

**Issue by godbit** [Lapsang-boys/Rex-Temporis#80]
Something like: https://www.hiveworkshop.com/threads/missile.265370/ [1]

I feel like one of our biggest problems right now is that the native spells are buggy and hard to use which is limiting us in our spell creation. A system like this where we could trigger everything ourselves would help a great deal and improve the spells used by mainly towers but (maybe) mobs as well. 

Suggested way of implementation:
Copy [1] and rewrite in wurst, maybe change things that we dont like

**Comment by karlek** [comment]
I agree that abstracting ourselves from the Warcraft 3 logic by using spell libraries would make our lives so much easier. There is a flavor of Wurst called Jurst which is specifically made for porting libraries to wurst. [1]

\[1\]: https://wurstlang.org/tutorials/legacymaps.html

**Comment by godbit** [comment]
Sweet, i did not know that. Although in the long run it might be worth to make the entire lib in wurst for easier development of extending classes, additional methods etc. And hopefully easier usage.
